language/stdlib/modules/validator_config_2.move

```diff
@@ -0,0 +1,97 @@
+address 0x0:
+
+module ValidatorConfig2 {
+    use 0x0::LibraAccount;
+    use 0x0::Transaction;
+
+    struct Config {
+        consensus_pubkey: vector<u8>,
+    }
+
+    // A current or prospective validator should publish one of these under their address.
+    // If delegation is enabled, only the delegated_address can change the config.
+    // If delegation is disabled, only the owner of this resource can change the config.
+    // The entity that can change the config is called a "validator operator".
+    // The owner of this resource can enable delegation and set delegated address.
+    // The owner can also disable delegation at any time.
+    resource struct T {
+        config: Config,
+        delegation_enabled: bool,
+        delegated_address: address,
+    }
+
+    // TODO(valerini): add events here
+
+    // Returns true if addr has a published ValidatorConfig::T resource
+    public fun has(addr: address): bool { exists<T>(addr) }
+
+    // Getters for Config
+    public fun get_config(addr: address): Config acquires T { *&borrow_global<T>(addr).config }
+    public fun get_consensus_pubkey(config_ref: &Config): vector<u8> { *&config_ref.consensus_pubkey }
+
+    // Returns the address of the entity who is now in charge of managing the config.
+    public fun get_validator_operator_address(addr: address): address acquires T {
+        let t_ref = borrow_global<T>(addr);
+        if (t_ref.delegation_enabled) {
+            return *&t_ref.delegated_address
+        } else {
+            return *&addr
```

Comment:
```suggestion
            addr
```

Reply:
> It would be fine to initialize this after genesis if it were anything other than the validator set... But we can certainly set it up so it can be manipulated after genesis. What do you have in mind?

I inherited this comment from the current [libra_system.move:53](https://github.com/libra/libra/blob/eb544c4ca94958ed2be4e629462d00b5276951dc/language/stdlib/modules/libra_system.move#L53). I don't think we need to manipulate it after genesis and initializing it in genesis is fine with me!